src/transaction.js

```diff
@@ -121,8 +121,15 @@ export default class Transaction extends EventEmitter {
         debug('%s error running transaction query', this.txid)
       })
       .tap(() => {
-        if (status === 1) this._resolver(value)
-        if (status === 2) this._rejecter(value)
+        if (status === 1) {
+          this._resolver(value)
+        }
+        if (status === 2) {
+          if(isUndefined(value)) {
```

Comment:
Yes, this is intentional. By definition `undefined` is the only javascript type that can be considered non-existant. Null, Strings, Numbers, Objects, Errors, Buffers, or any other type of value is still valid.

For example some people prefer to use strings (why is beyond me):
````javascript
try {
    throw 'poop'
} catch(e) {
    console.log(e, typeof e);
    //poop string
}
```

So in cases where rollback is called with an explicit value other than `undefined`, that value will be used in the reject handler.